crates/bevy_ecs/src/system/query.rs

```diff
@@ -772,29 +789,54 @@ where
         &self,
         entity: Entity,
     ) -> Result<Mut<'_, T>, QueryComponentError> {
-        let world = self.world;
-        let entity_ref = world
-            .get_entity(entity)
+        let (component_id, write_flag) = self
+            .state
+            .fetch_state
+            .get_id::<T>()
+            .ok_or(QueryComponentError::MissingWriteAccess)?;
+
+        if write_flag != RWAccess::Write {
```

Comment:
Can we add a comment or something that its ok that `get_id` returns immediately on finding `T` as if there is a `RWAccess::Write` there wont be any other query params for T. e.g. a query like `Query<(&T, &mut T)>` cant exist so we dont care that `get_id` would return `RwAccess::Read`